### PR TITLE
convert buildSummary.php to AngularJS

### PR DESCRIPTION
**Pull request by zackgalbreath**
Also fix the charts for subproject builds (they were showing dots for unrelated builds).

### CODE REVIEW

**Comment by zackgalbreath** [comment]
@BetsyMcPhail Since you were the most recent person working on this page, would you PTAL and make sure everything still works as intended?

File: public/js/controllers/buildSummary.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+      var buildwarnings = [];
+      var testfailed = [];
+      var buildids = [];
+
+      $scope.cdash.buildhistory = [];
+
+      // Isolate data for each graph.
+      for (var i = 0, len = builds.length; i < len; i++) {
+        var build = builds[i];
+        var t = new Date(build['starttime']).getTime();
```
Comment (BetsyMcPhail) [inline public/js/controllers/buildSummary.js]: This returns 'NaN' on my system. 

var t = new Date("2016-09-08 16:09:20").getTime();  <---- NaN

var t = new Date("2016-09-08T16:09:20").getTime();  <--- Ok

**Comment by BetsyMcPhail** [comment]
Nothing is displayed on my graphs due to Date().getTime() returning NaN!

I also noticed that the 'Zoom out' control is missing, so I can zoom into a graph with the mouse but can't get back to where I started?

**Comment by zackgalbreath** [comment]
Thanks for the feedback, Betsy.  The NaN issue should not happen anymore.  You can now reset the zoom level on the graphs by double clicking on them.

Please take another look when you get some time & let me know if you find any other problems.

**Comment by BetsyMcPhail** [comment]
Two more small issues:
1. The 'ViewErrorsSummary' link doesn't populate the build id (the other 'View...' links are okay)
2. The 'Configure', 'Build', and 'Test' links under 'Current Build' are broken

**Comment by zackgalbreath** [comment]
Thanks again for the thorough review, @BetsyMcPhail.  These latest issues that you discovered should be fixed now.

**Comment by BetsyMcPhail** [comment]
LGTM!

**Pull request merged**